File: code/3768字符串删减.py
if __name__ == "__main__":
    n = int(input())
    s = input()

    res, cnt = 0, 0
    for i in range(n):
        if s[i] == 'x':
            cnt += 1
        else:
            cnt = 0
        if cnt >= 3:
            res += 1
            cnt -= 1
    print(res)

File: code/3768字符串删减.cpp
#include <algorithm>
#include <cstring>
#include <iostream>
using namespace std;

int main() {
  int n, res = 0, cnt = 0;
  string str;

  cin >> n >> str;

  for (int i = 0; i < str.size(); i++) {
    if (str[i] == 'x')
      cnt++;
    else
      cnt = 0;
    if (cnt >= 3)
      cnt -= 1, res += 1;
  }

  cout << res << endl;

  return 0;
}
